src/main/java/hu/oe/nik/szfmv/automatedcar/AutomatedCar.java

```diff
@@ -51,12 +49,95 @@ public PowerTrain getPowerTrain() {
+        this.setX((int)position.getXDiff());
+        this.setY((int)position.getYDiff());
+    }
+
+    private void updatePositionAndOrientation() {
+        IVector carMove = virtualFunctionBus.carPositionPacket.getMoveVector();
+        this.moveCar(carMove);
+    }
+
+    /**Applies movement to the car.
+     * @param move The movement to apply.
+     * <p>- its direction is interpreted as facing direction of the front wheels.</p>
+     * <p>- its length is interpreted as the speed of movement.</p>*/
+    private void moveCar(IVector move) {
+        switch (powerTrain.transmission.getGearMode()) {
+            case D_DRIVE:
+                this.moveForward(move);
+                break;
+            case R_REVERSE:
+                this.moveBackward(move);
+                break;
+            case N_NEUTRAL:
+                this.neturalGear();
+                break;
+            default:
+                this.parkingGear();
+                break;
+        }
+    }
+
+    private void moveForward(IVector forwardMove) {
+        IVector currentPosition = this.getPosition();
+        IVector toCarFrontVector = this.facingDirection.withLength(getHeight() / 2.0);
+        IVector carFrontPosition = currentPosition.add(toCarFrontVector);
+        IVector carBackPosition = currentPosition.subtract(toCarFrontVector);
+
+        IVector moveInDirection = forwardMove.rotateByRadians(this.facingDirection.getRadians());
+
+        IVector newCarFrontPosition = carFrontPosition.add(moveInDirection);
+        IVector newCarBackPosition = carBackPosition.add(forwardMove.withDirection(toCarFrontVector));
+
+        this.setPosition(average(newCarFrontPosition, newCarBackPosition));
+
+        this.facingDirection = newCarFrontPosition.subtract(newCarBackPosition);
+        this.setRotation(facingDirection.getRadiansRelativeTo(Axis.Y.negativeDirection()));
+    }
+
+    private void moveBackward(IVector backwardMove) {
+        IVector currentPosition = this.getPosition();
+        IVector toCarBackVector = this.facingDirection.withLength(getHeight() / -2.0);
+        IVector carFrontPosition = currentPosition.subtract(toCarBackVector);
+        IVector carBackPosition = currentPosition.add(toCarBackVector);
+
+        IVector moveInDirection = backwardMove.rotateByRadians(this.facingDirection.getRadians()).multiplyBy(-1);
+
+        IVector newCarFrontPosition = carFrontPosition.add(moveInDirection);
+        IVector newCarBackPosition = carBackPosition.add(backwardMove.withDirection(toCarBackVector));
+
+        this.setPosition(average(newCarFrontPosition, newCarBackPosition));
+
+        this.facingDirection = newCarFrontPosition.subtract(newCarBackPosition);
+        this.setRotation(facingDirection.getRadiansRelativeTo(Axis.Y.negativeDirection()));
+    }
+
+    private void neturalGear() {
+        //FAAAAAAAKE!!!
+    }
+
+    private void parkingGear() {
+        //FAAAAAKE as FAF, but it will be probably enough for parking mode.
```

Comment:
Sosem vezettem automata váltós autót, de ez lehet csak annyi, hogy P-ben nem lehet megmozdítani az autót, akár ilyen motor leállítva állapotnak is lehet kezelni.